include/boost/utility/string_ref.hpp

```diff
@@ -424,48 +425,63 @@ namespace boost {
     namespace detail {
 
         template<class charT, class traits>
-        inline void sr_insert_fill_chars(std::basic_ostream<charT, traits>& os, std::size_t n) {
-            enum { chunk_size = 8 };
-            charT fill_chars[chunk_size];
-            std::fill_n(fill_chars, static_cast< std::size_t >(chunk_size), os.fill());
-            for (; n >= chunk_size && os.good(); n -= chunk_size)
-                os.write(fill_chars, static_cast< std::size_t >(chunk_size));
-            if (n > 0 && os.good())
-                os.write(fill_chars, n);
+        inline bool sr_os_fill(std::basic_ostream<charT, traits>& os, std::size_t count)
+        {
+            for (charT c = os.fill(); count > 0; --count) {
+                if (traits::eq_int_type(os.rdbuf()->sputc(c), traits::eof())) {
+                    return false;
+                }
             }
+            return true;
+        }
 
         template<class charT, class traits>
-        void sr_insert_aligned(std::basic_ostream<charT, traits>& os, const basic_string_ref<charT,traits>& str) {
-            const std::size_t size = str.size();
-            const std::size_t alignment_size = static_cast< std::size_t >(os.width()) - size;
-            const bool align_left = (os.flags() & std::basic_ostream<charT, traits>::adjustfield) == std::basic_ostream<charT, traits>::left;
-            if (!align_left) {
-                detail::sr_insert_fill_chars(os, alignment_size);
-                if (os.good())
-                    os.write(str.data(), size);
-                }
-            else {
-                os.write(str.data(), size);
-                if (os.good())
-                    detail::sr_insert_fill_chars(os, alignment_size);
+        class sr_os_holder {
+        public:
+            explicit sr_os_holder(std::basic_ostream<charT, traits>& os)
+                : os_(&os) { }
+            ~sr_os_holder() {
+                if (os_) {
+                    os_->setstate(std::basic_ostream<charT, traits>::badbit);
```

Comment:
If this throws, it will call `terminate` since the destructor is `noexcept` by default.

Reply:
Corrected noexcept specifier on destructor.